api/invite.go

```diff
@@ -526,3 +694,338 @@ func (a *Api) SendInvite(res http.ResponseWriter, req *http.Request, vars map[st
 	}
 	return
 }
+
+// @Summary Send invitation to a hcp joining a medical team
+// @Description  create a notification for the invitee and send him an email with the invitation
+// @ID hydrophone-api-SendTeamInvite
+// @Accept  json
+// @Produce  json
+// @Success 200 {object} models.Confirmation "invite details"
+// @Failure 400 {object} status.Status "userId, teamId and isAdmin were not provided or the payload is missing/malformed"
+// @Failure 401 {object} status.Status "Authorization token is missing or does not provide sufficient privileges"
+// @Failure 403 {object} status.Status "Authorization token is invalid"
+// @Failure 409 {object} status.Status "user already has a pending or declined invite OR user is already part of the team"
+// @Failure 422 {object} status.Status "Error when sending the email (probably caused by the mailling service"
+// @Failure 500 {object} status.Status "Internal error while processing the invite, detailled error returned in the body"
+// @Router /confirm/send/team-invite [post]
+// @security TidepoolAuth
+func (a *Api) SendTeamInvite(res http.ResponseWriter, req *http.Request, vars map[string]string) {
+	// By default, the invitee language will be "en" for Englih (as we don't know which language suits him)
+	// In case the invitee is a known user, the language will be overriden in a later step
+	var inviteeLanguage = "en"
+	if token := a.token(res, req); token != nil {
+
+		invitorID := token.UserID
+		if invitorID == "" {
+			res.WriteHeader(http.StatusBadRequest)
+			return
+		}
+
+		defer req.Body.Close()
+		var ib = &inviteBody{}
+		if err := json.NewDecoder(req.Body).Decode(ib); err != nil {
+			log.Printf("SendInvite: error decoding invite to detail %v\n", err)
+			statusErr := &status.StatusError{Status: status.NewStatus(http.StatusBadRequest, STATUS_ERR_DECODING_INVITE)}
+			a.sendModelAsResWithStatus(res, statusErr, statusErr.Code)
+			return
+		}
+
+		if ib.Email == "" || ib.TeamID == "" || ib.IsAdmin == "" {
+			statusErr := &status.StatusError{Status: status.NewStatus(http.StatusBadRequest, STATUS_ERR_MISSING_DATA_INVITE)}
+			a.sendModelAsResWithStatus(res, statusErr, statusErr.Code)
+			return
+		}
+
+		if auth, err := a.tokenUserIsTeamAdmin(token, ib.TeamID); !auth || err != nil {
+			statusErr := &status.StatusError{Status: status.NewStatus(http.StatusUnauthorized, STATUS_NOT_ADMIN)}
+			a.sendModelAsResWithStatus(res, statusErr, statusErr.Code)
+			return
+		}
+
+		// TODO get the team Name from teams service
+		teamName := "Name of the team"
+		teamAddress := "Address of the team"
+		teamIdentification := "Shared ID of the team"
+
+		var sessionToken = req.Header.Get(TP_SESSION_TOKEN)
+		// check duplicate invite and if user is already a member
+		if existingInvite, invitedUsr := a.checkForDuplicateTeamInvite(req.Context(), ib.Email, invitorID, ib.TeamID, sessionToken, models.TypeMedicalTeamInvite, res); existingInvite == true {
```

Comment:
If we use crew.GetTeam("teamID") above to get the team details, we get the list of members. This can be used to: 
1. check that the invitor is effectively an admin
2. check the the invitee is not already part of the members

This can avoid other calls to Crew

Reply:
updated in recent commits